src/main/java/teammates/ui/webapi/GetInstructorsAction.java

```diff
@@ -28,69 +31,147 @@ void checkSpecificAccessControl() throws UnauthorizedAccessException {
-            // adds googleId if caller is admin or has the appropriate privilege to modify instructor
-            if (userInfo.isAdmin || logic.getInstructorForGoogleId(courseId, userInfo.getId()).getPrivileges()
-                    .isAllowedForPrivilege(Const.InstructorPermissions.CAN_MODIFY_INSTRUCTOR)) {
+        if (isCourseMigrated(courseId)) {
+            List<Instructor> instructorsOfCourse = sqlLogic.getInstructorsByCourse(courseId);
+
+            if (intentStr == null) {
+                instructorsOfCourse = instructorsOfCourse
+                        .stream()
+                        .filter(Instructor::isDisplayedToStudents)
+                        .collect(Collectors.toList());
                 data = new InstructorsData();
-                for (InstructorAttributes instructor : instructorsOfCourse) {
-                    InstructorData instructorData = new InstructorData(instructor);
-                    instructorData.setGoogleId(instructor.getGoogleId());
-                    if (userInfo.isAdmin) {
-                        instructorData.setKey(instructor.getKey());
+
+                List<InstructorData> instructorDataList = instructorsOfCourse
+                        .stream()
+                        .map(InstructorData::new)
+                        .collect(Collectors.toList());
+
+                data.setInstructors(instructorDataList);
+
+                // hide information
+                data.getInstructors().forEach(i -> {
+                    i.setGoogleId(null);
+                    i.setJoinState(null);
+                    i.setIsDisplayedToStudents(null);
+                    i.setRole(null);
+                });
+            } else if (intentStr.equals(Intent.FULL_DETAIL.toString())) {
+                // get all instructors of a course without information hiding
+                // adds googleId if caller is admin or has the appropriate privilege to modify instructor
+                if (userInfo.isAdmin || sqlLogic.getInstructorByGoogleId(courseId, userInfo.getId()).getPrivileges()
+                        .isAllowedForPrivilege(Const.InstructorPermissions.CAN_MODIFY_INSTRUCTOR)) {
+                    data = new InstructorsData();
+
+                    for (Instructor instructor : instructorsOfCourse) {
+                        InstructorData instructorData = new InstructorData(instructor);
+                        instructorData.setGoogleId(instructor.getAccount().getGoogleId());
+                        if (userInfo.isAdmin) {
+                            instructorData.setKey(instructor.getRegKey());
+                        }
+                        data.getInstructors().add(instructorData);
                     }
-                    data.getInstructors().add(instructorData);
+                } else {
+                    data = new InstructorsData();
+
+                    List<InstructorData> instructorDataList = instructorsOfCourse
```

Comment:
Every time you do something that is not idiomatic you should add comments do document why this was the case.

Reply:
Thanks samuel, I should and will in the future! Might not be necessary, have added some comments to the non-migrated code after changing `InstructorsData`.